images/gh-gl-sync/SpackCIBridge.py

```diff
@@ -536,14 +567,21 @@ def sync(self):
             self.setup_git_repo()
 
             if self.main_branch:
-                # Find the currently running main branch pipeline, if any, and get the sha
+                # Find the currently running main branch pipeline, if any, and get the sha.
+                # Also get the latest commit on the main branch that has a completed pi
                 main_branch_pipelines = self.get_pipelines_for_branch(self.main_branch, self.time_threshold_long)
                 for sha, pipeline in main_branch_pipelines.items():
+                    if self.latest_tested_main_commit is None and \
+                            (pipeline['status'] == "success" or pipeline['status'] == "failed"):
+                        self.latest_tested_main_commit = sha
+
                     if pipeline['status'] == "running":
```

Comment:
Initially I didn't understand why we wouldn't need here a similar guard to the one above, where we make sure we don't set `self.currently_running_sha` if it's already set.  I guess maybe it's because this whole thing hinges on there only ever being a single running main branch pipeline at a time?

Reply:
doesn't hurt to be safe, I'll add this check